db/migrate/20220204211210_add_user_ref_to_posts.rb

```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :users, null: false, foreign_key: true
```

Comment:
Please add `author_id` as a foreign key instead of `user_id` and reference it to the User model.